frontend/src/components/CodeEditorV2.tsx

```diff
@@ -0,0 +1,130 @@
+import { EditorState, type Extension } from '@codemirror/state'
+import { EditorView, drawSelection, gutter, highlightActiveLineGutter, hoverTooltip, keymap, lineNumbers } from '@codemirror/view'
+
+import { bracketMatching, defaultHighlightStyle, foldGutter, foldKeymap, indentOnInput, syntaxHighlighting } from '@codemirror/language'
+import { lintGutter } from '@codemirror/lint'
+import { lintKeymap } from '@codemirror/lint'
+import { linter } from '@codemirror/lint'
+
+import { autocompletion, closeBrackets, closeBracketsKeymap } from '@codemirror/autocomplete'
+import { atomone } from '@uiw/codemirror-theme-atomone'
+import { githubLight } from '@uiw/codemirror-theme-github'
+
+import { defaultKeymap, history, indentWithTab, redo, undo } from '@codemirror/commands'
+import { handleRefresh, jsonSchemaHover, jsonSchemaLinter, stateExtensions } from 'codemirror-json-schema'
+import { json5, json5ParseLinter } from 'codemirror-json5'
+import { useEffect, useRef, useState } from 'react'
+import { useUserPreferences } from '../providers/user-preferences-provider'
+
+export const CodeEditor = ({
```

Comment:
Does it make sense to just ditch the old editor and replace the usages with this one instead?

Reply:
Sadly they handle the state slightly differently, so it'd be a more time consuming refactor than is worth it :/